src/app/components/quick-pick/quick-pick.component.ts

```diff
@@ -0,0 +1,163 @@
+import {FormGroup, FormControl, Validators} from "@angular/forms";
+import {ExpressionInputComponent} from "../forms/inputs/types/expression-input.component";
+import {ExpressionModel} from "cwlts/models/d2sb";
+import {BehaviorSubject, Subscription, ReplaySubject} from "rxjs";
+import {ExpressionSidebarService} from "../../services/sidebars/expression-sidebar.service";
+
+require("./quick-pick.component.scss");
+
+@Component({
+    selector: "quick-pick",
+    directives: [
+        ExpressionInputComponent
+    ],
+    template: `
+        <form [formGroup]="formGroup"
+             class="button-list content-wrapper"
+             *ngIf="!formGroup.controls['expressionField'] && formGroup.controls['radioButtonControl']">
+        
+           <div class="btn-group" data-toggle="buttons">
+                
+                <label *ngFor="let buttonLabel of buttonList" 
+                       [ngClass]="{'active': buttonLabel === formGroup.controls['radioButtonControl'].value}"
+                       class="btn btn-secondary button-item-label">
+                       
+                        <input type="radio"
+                               formControlName="radioButtonControl"
+                               [value]="buttonLabel">
+                               {{buttonLabel}}
+                </label>
+          </div>
+         
+          <button type="button" 
+                  class="btn btn-primary add-expression-btn"
+                  *ngIf="!formGroup.controls['expressionField']"
+                  (click)="crateExpressionInput()">Add Expression</button>
+        </form>
+        
+        
+        <div class="expression-input-wrapper content-wrapper"
+             *ngIf="formGroup.controls['expressionField']">
+            
+            <expression-input class="col-sm-11 expression-input"
+                      [control]="formGroup.controls['expressionField']"
+                      [isExpression]="!!expressionModel.serialize().script"
+                      (onEdit)="editExpression()"
+                      (onClear)="clearExpression()">
+            </expression-input>
+            
+            <span class="col-sm-1">
+                <i class="fa fa-trash clear-icon" (click)="removeExpressionInput()"></i>
+            </span>
+        </div>
+    `
+})
+export class QuickPickComponent {
+
+    @Input()
+    public buttonList: String[];
+
+    @Input()
+    public context: any;
+
+    @Input()
+    private formGroup: FormGroup;
+
+    @Output()
+    public onUpdate = new ReplaySubject<string>(1);
+
+    private expressionModel: ExpressionModel;
+
+    private expressionInputSub: Subscription;
+
+    private subs: Subscription[] = [];
+
+    constructor(private expressionSidebarService: ExpressionSidebarService) {
+
+    }
+
+    ngOnInit() {
+        this.formGroup.addControl(
+            "radioButtonControl", new FormControl("")
+        );
+
+        this.subs.push(
```

Comment:
You can extend the `ComponentBase` class and assign subscriptions and other objects that should be destroyed to the `tracked` property.  If you're implementing the `ngOnDestroy` method, be sure to call `super.ngOnDestroy` inside it.